File: examples/ruby/ping_subnet.rb
#!/usr/bin/ruby

done = []
ip_s = []
range = 1..254

range.each do |i|
  ip_address = "10.11.1.#{i}"
  ping = `ping -q -c 1 -W 3 #{ip_address}`.split(/\n+/)[2].split(/,/)[1].split(/ /)[1].to_i
  
  puts "#{ip_address} #{ping}"

end

File: examples/perl/ping_subnet.pl
#!/usr/bin/perl

use strict;
use warnings;

use Net::Ping;

my $ip_base = "10.11.1.";
my @range = (1..254);

my $p = Net::Ping->new();
for (@range){
	my $ip =join "", ${ip_base}, ${_};
	my $return = $p->ping($ip);
	if( $return == 0 ){
		print("${ip} 1\n");
	}else{
		print("${ip} 0\n");
	}
}
$p->close();
